### PR TITLE
soc: sam0: samd5x: xosc32 configurable startup time

### LINKED ISSUE #83147: soc: samd5x: xosc32: Allow users to select the start-up time
**Is your enhancement proposal related to a problem? Please describe.**
The Atmel samd5x have a fixed start-up time for the xosc32. Provide a way to select the start-up time for the xosc32 from samd5x platform.

**Describe alternatives you've considered**
* Kconfig: it is the easy way to implement.
* devicetree: require high effort becuase platform don't provide clock drivers yet.

### PR DESCRIPTION
**Pull request by wag-thom**
Adds Kconfig option to configure the startup time of the external 32KHz crystal oscillator.

As discussed in https://github.com/zephyrproject-rtos/zephyr/discussions/79883

### CODE REVIEW

File: soc/atmel/sam0/common/Kconfig.samd5x [hunk 1]
```diff
@@ -10,6 +10,14 @@ config SOC_ATMEL_SAMD5X_XOSC32K
 	  startup.  This can then be selected as the main clock source
 	  for the SOC.
 
+config SOC_ATMEL_SAMD5X_XOSC32K_STARTUP
+	depends on SOC_ATMEL_SAMD5X_XOSC32K
+	hex "Startup time external 32 kHz crystal oscillator"
+	range 0x0 0x6
+	default 0x0
```
Comment (nandojve) [inline soc/atmel/sam0/common/Kconfig.samd5x:17]: set default to 0x1

File: soc/atmel/sam0/common/soc_samd5x.c [hunk 2]
```diff
@@ -23,8 +23,9 @@
 static void osc32k_init(void)
 {
 	OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_ENABLE | OSC32KCTRL_XOSC32K_XTALEN
-				| OSC32KCTRL_XOSC32K_EN32K | OSC32KCTRL_XOSC32K_RUNSTDBY
-				| OSC32KCTRL_XOSC32K_STARTUP(0) | OSC32KCTRL_XOSC32K_CGM_XT;
+			| OSC32KCTRL_XOSC32K_EN32K | OSC32KCTRL_XOSC32K_RUNSTDBY
+			| OSC32KCTRL_XOSC32K_STARTUP(CONFIG_SOC_ATMEL_SAMD5X_XOSC32K_STARTUP)
+			| OSC32KCTRL_XOSC32K_CGM_XT;
```
Comment (nandojve) [inline soc/atmel/sam0/common/soc_samd5x.c:28]: I'll ask you to keep below order and code style.

```diff
-	OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_ENABLE | OSC32KCTRL_XOSC32K_XTALEN
-				| OSC32KCTRL_XOSC32K_EN32K | OSC32KCTRL_XOSC32K_RUNSTDBY
-				| OSC32KCTRL_XOSC32K_STARTUP(0) | OSC32KCTRL_XOSC32K_CGM_XT;
+	OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_ENABLE
+				| OSC32KCTRL_XOSC32K_XTALEN
+				| OSC32KCTRL_XOSC32K_CGM_XT
+				| OSC32KCTRL_XOSC32K_EN32K
+				| OSC32KCTRL_XOSC32K_RUNSTDBY
+				| OSC32KCTRL_XOSC32K_STARTUP(CONFIG_SOC_ATMEL_SAMD5X_XOSC32K_STARTUP)
+				;
```

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @wag-thom,

Nice to see you came back to improve this : )

There are new changes already programmed for this topic. This is why I'm asking to you small changes.

File: soc/atmel/sam0/common/soc_samd5x.c (hunk 2 above)
Comment (nandojve) [reply soc/atmel/sam0/common/soc_samd5x.c:28]: Yes, it matters.

You can do something like:
```diff
#define SAM0_DPLL_FREQ_MAX_HZ		(200000000U)
+#define SAM0_XOSC32K_STARTUP_TIME	CONFIG_SOC_ATMEL_SAMD5X_XOSC32K_STARTUP

#if CONFIG_SOC_ATMEL_SAMD5X_XOSC32K_AS_MAIN
static void osc32k_init(void)
{
+	OSC32KCTRL->XOSC32K.reg = OSC32KCTRL_XOSC32K_ENABLE
+				| OSC32KCTRL_XOSC32K_XTALEN
+				| OSC32KCTRL_XOSC32K_CGM_XT
+				| OSC32KCTRL_XOSC32K_EN32K
+				| OSC32KCTRL_XOSC32K_RUNSTDBY
+				| OSC32KCTRL_XOSC32K_STARTUP(SAM0_XOSC32K_STARTUP_TIME)
+				;
```

**Review by nandojve — APPROVED** [review]

**Comment by nandojve** [comment]
Changes LGTM.

Just make sure CI is green. You can run `scripts/ci/check_compliance.py` to check by problems.

**Review by pdgendt — APPROVED** [review]

**Pull request merged**